packages/components/src/DataTable/DataTable.tsx

```diff
@@ -0,0 +1,186 @@
+
+  /**
+   * Enables row click action. The provided callback will be executed when the row is clicked.
+   */
+  onRowClick?: (row: Row<T>) => void;
+}
+
+export function DataTable<T extends object>({
+  data,
+  columns,
+  pagination,
+  sorting,
+  height,
+  stickyHeader,
+  pinFirstColumn,
+  onRowClick,
+}: DataTableProps<T>) {
+  const tableSettings = createTableSettings(data, columns, {
+    pagination,
+    sorting,
+  });
+
+  const stickyClass = classNames({ [styles.stickyHeader]: stickyHeader });
+
+  const tableClasses = classNames(styles.table, {
+    [styles.pinFirstColumn]: pinFirstColumn,
+  });
+
+  const bodyRowClasses = classNames({ [styles.clicableRow]: !!onRowClick });
+
+  const handleRowClick = useCallback(
+    (row: Row<T>) => () => {
+      if (onRowClick == undefined) return;
+      onRowClick(row);
+    },
+    [onRowClick],
+  );
+
+  const table = useReactTable(tableSettings);
+  return (
+    <div className={styles.dataTableContainer}>
+      <div className={styles.tableContainer} style={{ height }}>
+        <table className={tableClasses}>
+          <thead className={stickyClass}>
+            {table.getHeaderGroups().map(headerGroup => (
+              <tr key={headerGroup.id}>
+                {headerGroup.headers.map(header => {
+                  return (
+                    <th
+                      key={header.id}
+                      colSpan={header.colSpan}
+                      className={
+                        sorting && header.column.getCanSort()
+                          ? classNames(styles.sortableColumn, {
+                              [styles.pinFirstHeaderSortable]: !!onRowClick,
+                            })
+                          : ""
+                      }
+                      onClick={
+                        sorting
+                          ? header.column.getToggleSortingHandler()
+                          : undefined
+                      }
+                    >
+                      {header.isPlaceholder ? null : (
+                        <div>
+                          {flexRender(
+                            header.column.columnDef.header,
+                            header.getContext(),
+                          )}
+                          {{
+                            asc: <Icon name="longArrowUp" color="green" />,
+                            desc: <Icon name="longArrowDown" color="green" />,
+                          }[header.column.getIsSorted() as string] ?? (
+                            <svg
+                              style={{
+                                width: "var(--space-large)",
+                                height: "var(--space-large)",
+                              }}
+                            />
+                          )}
+                        </div>
+                      )}
+                    </th>
+                  );
+                })}
+              </tr>
+            ))}
+          </thead>
```

Comment:
As we have the `Footer` component, is it makes sense to turn this part into a `Header` component?

Reply:
I believe that you're correct, thanks for the suggestion.

Reply:
✅